lib/elixir/lib/keyword.ex

```diff
@@ -69,9 +69,7 @@ defmodule Keyword do
   """
   @spec new(Enum.t) :: t
   def new(pairs) do
-    Enum.reduce pairs, [], fn {k, v}, keywords ->
-      put(keywords, k, v)
-    end
+    Enum.uniq_by(Enum.reverse(pairs), fn {x, _} when is_atom(x) -> x end)
```

Comment:
The old one is definitely more performant for smaller sets (but that is looking at the algorithm complexity). If you bench and verify it is actually true, it would be a really welcome sanity check!